diff --git a/drivers/staging/media/atomisp/pci/Kconfig b/drivers/staging/media/atomisp/pci/Kconfig
index a72421431c7a..e3e00ade1d38 100644
--- a/drivers/staging/media/atomisp/pci/Kconfig
+++ b/drivers/staging/media/atomisp/pci/Kconfig
@@ -11,3 +11,26 @@  config VIDEO_ATOMISP
           camera imaging subsystem.
           To compile this driver as a module, choose M here: the
           module will be called atomisp
+
+choice
+        prompt "Intel Atom Image Signal Processor Driver Type"
+        depends on VIDEO_ATOMISP
+        default VIDEO_ATOMISP_ISP2400
+        help
+          Intel Atom Image Signal Processor Driver actually doesn't support
+          dynamically all SoC.
+          So need to choose at compilation time which SoC it can support.
+          Please refer to staging TODO for more details.
+
+config VIDEO_ATOMISP_ISP2400
+        bool "ISP2400"
+        help
+          Atom ISP for Merrifield, Baytrail SoC.
+
+config VIDEO_ATOMISP_ISP2401
+        bool "ISP2401"
+        help
+          Atom ISP for Anniedale (Merrifield+ / Moorefield), Cherrytrail SoC.
+
+endchoice
+
diff --git a/drivers/staging/media/atomisp/pci/atomisp2/Makefile b/drivers/staging/media/atomisp/pci/atomisp2/Makefile
index 2bd98f0667ec..27ac23c0c18d 100644
--- a/drivers/staging/media/atomisp/pci/atomisp2/Makefile
+++ b/drivers/staging/media/atomisp/pci/atomisp2/Makefile
@@ -155,7 +155,7 @@  atomisp-objs += \
 	hmm/hmm_dynamic_pool.o \
 	hrt/hive_isp_css_mm_hrt.o \
 	atomisp_v4l2.o
-	
+
 # These will be needed when clean merge CHT support nicely into the driver
 # Keep them here handy for when we get to that point
 #
@@ -347,8 +347,16 @@  DEFINES := -DHRT_HW -DHRT_ISP_CSS_CUSTOM_HOST -DHRT_USE_VIR_ADDRS -D__HOST__
 #DEFINES += -DPUNIT_CAMERA_BUSY
 #DEFINES += -DUSE_KMEM_CACHE
 
+ifeq ($(CONFIG_VIDEO_ATOMISP_ISP2400),y)
+# Merrifield, Baytrail
 DEFINES += -DATOMISP_POSTFIX=\"css2400b0_v21\" -DISP2400B0
 DEFINES += -DSYSTEM_hive_isp_css_2400_system -DISP2400
+endif
+ifeq ($(CONFIG_VIDEO_ATOMISP_ISP2401),y)
+# Anniedale (Merrifield+ / Moorefield), Cherrytrail
+DEFINES += -DATOMISP_POSTFIX=\"css2401a0_v21\" -DISP2401A0
+DEFINES += -DSYSTEM_hive_isp_css_2401_system -DISP2401 -DISP2401_NEW_INPUT_SYSTEM
+endif
 
 ccflags-y += $(INCLUDES) $(DEFINES) -fno-common
 
