diff --git a/drivers/staging/media/atomisp/pci/atomisp2/css2400/ia_css_acc_types.h b/drivers/staging/media/atomisp/pci/atomisp2/css2400/ia_css_acc_types.h
index a2a1873aca83..e457f1190b62 100644
--- a/drivers/staging/media/atomisp/pci/atomisp2/css2400/ia_css_acc_types.h
+++ b/drivers/staging/media/atomisp/pci/atomisp2/css2400/ia_css_acc_types.h
@@ -319,9 +319,10 @@ struct ia_css_sp_info {
 	uint32_t host_sp_queues_initialized; /**< Polled from the SP */
 	uint32_t sleep_mode;  /**< different mode to halt SP */
 	uint32_t invalidate_tlb;		/**< inform SP to invalidate mmu TLB */
-#ifndef ISP2401
+//#ifndef ISP2401
+        //FIXME CHECK - SEEMS ifndef is wrong from Aero tree #ifndef ISP2401
 	uint32_t stop_copy_preview;       /**< suspend copy and preview pipe when capture */
-#endif
+//#endif
 	uint32_t debug_buffer_ddr_address;	/**< inform SP the address
 	of DDR debug queue */
 	uint32_t perf_counter_input_system_error; /**< input system perf
