diff --git a/drivers/staging/media/atomisp/pci/atomisp2/css2400/sh_css_firmware.c b/drivers/staging/media/atomisp/pci/atomisp2/css2400/sh_css_firmware.c
index 8158ea40d069..de9e289cbc8a 100644
--- a/drivers/staging/media/atomisp/pci/atomisp2/css2400/sh_css_firmware.c
+++ b/drivers/staging/media/atomisp/pci/atomisp2/css2400/sh_css_firmware.c
@@ -185,7 +185,10 @@ sh_css_check_firmware_version(const char *fw_data)
 	file_header = &firmware_header->file_header;
 
 	if (strcmp(file_header->version, release_version) != 0) {
-		return false;
+		printk("Warning: atomisp CSS code version (%s) and firmware version (%s) mismatch! Trying to load it anyway",
+				file_header->version, release_version);
+		//return false;
+		return true;
 	} else {
 		/* firmware version matches */
 		return true;
